### PR TITLE
[12.0] [MIG] Account Bank Statement Import CAMT OCA

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**

### CODE REVIEW

File: account_bank_statement_import_camt_oca/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# © 2013-2017 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/__manifest__.py]: © has to be replaced as per https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: account_bank_statement_import_camt_oca/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/__init__.py]: Please follow https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: account_bank_statement_import_camt_oca/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'CAMT Format Bank Statements Import',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Odoo Community Association (OCA), Therp BV',
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/__manifest__.py]: Please reorder as per https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: account_bank_statement_import_camt_oca/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# © 2013-2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/__manifest__.py]: Please use https

File: account_bank_statement_import_camt_oca/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/__init__.py]: No copyright needed here

File: account_bank_statement_import_camt_oca/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/__init__.py]: Please follow https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: account_bank_statement_import_camt_oca/models/account_bank_statement_import.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+"""Add process_camt method to account.bank.statement.import."""
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/account_bank_statement_import.py]: Please adjust copyright information

File: account_bank_statement_import_camt_oca/models/account_bank_statement_import.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
+"""Add process_camt method to account.bank.statement.import."""
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/account_bank_statement_import.py]: Is this docstring really needed?

File: account_bank_statement_import_camt_oca/models/parser.py [hunk 9]
```diff
@@ -0,0 +1,249 @@
+"""Class to parse camt files."""
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/parser.py:1]: Isn't it obvious?

File: account_bank_statement_import_camt_oca/models/parser.py [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/models/parser.py]: Please use copyright format as per https://github.com/OCA/maintainer-tools/tree/master/template/module

File: account_bank_statement_import_camt_oca/tests/__init__.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+"""Test import of bank statement for camt.053."""
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/tests/__init__.py]: Copyright, useless docstring

File: account_bank_statement_import_camt_oca/tests/test_import_bank_statement.py [hunk 12]
```diff
@@ -0,0 +1,152 @@
+"""Run test to import camt.053 import."""
+# © 2013-2016 Therp BV <http://therp.nl>
```
Comment (alexey-pelykh) [inline account_bank_statement_import_camt_oca/tests/test_import_bank_statement.py]: Copyright + docstring

File: setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca [hunk 13]
```diff
@@ -0,0 +1 @@
+../../../../account_bank_statement_import_camt_oca
```
Comment (alexey-pelykh) [inline setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca:1]: @pedrobaeza is this needed?

File: setup/account_bank_statement_import_camt_oca/setup.py [hunk 14]
```diff
@@ -0,0 +1,6 @@
+import setuptools
```
Comment (alexey-pelykh) [inline setup/account_bank_statement_import_camt_oca/setup.py:1]: As far as I know, `setup` should not be part of migration?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca (hunk 13 above)
Comment (pedrobaeza) [reply setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca:1]: Well, it's a kind of trick/requirement for pip deployment (for which the company of this contributor, Acsone, is the main promoter). It's added by the bot at merge time in the same way as README is generated from fragments, but can be pre-generated directly in the PR.

Comment (alexey-pelykh) [reply setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca:1]: Maybe let's keep commits clean from autogenerated? For future migrators?

Comment (pedrobaeza) [reply setup/account_bank_statement_import_camt_oca/odoo/addons/account_bank_statement_import_camt_oca:1]: The migration procedure rips-off this automatically, so no need to put it on a separate commit. I even prefer to be in the same.

**Comment by ThomasBinsfeld** [comment]
@alexey-pelykh Done

**Review by chrisdec — APPROVED** [review]
Functional test

**Review by alexey-pelykh — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@astirpe since you've contributed to this module as well, if possible, take a look on this PR

**Comment by alexey-pelykh** [comment]
@pedrobaeza is this module good to merge?

**Comment by sbidoul** [comment]
tested, merging

**Pull request merged**